source/backend/api/Services/ActivityService.cs

```diff
@@ -187,5 +192,27 @@ private void ValidateVersion(long activityId, long activityVersion)
                 throw new DbUpdateConcurrencyException("You are working with an older version of this activity, please refresh the application and retry.");
             }
         }
+
+        private void AddNoteIfActivityStatusChanged(PimsActivityInstance instance)
+        {
+            var currentActivity = _activityRepository.GetById(instance.Id);
+            if (currentActivity.ActivityInstanceStatusTypeCode != instance.ActivityInstanceStatusTypeCode)
+            {
+                PimsActivityInstanceNote note = new PimsActivityInstanceNote()
+                {
+                    ActivityInstanceId = currentActivity.ActivityInstanceId,
+                    AppCreateTimestamp = System.DateTime.Now,
+                    AppCreateUserid = instance.AppCreateUserid,
+                    Note = new PimsNote()
+                    {
+                        IsSystemGenerated = true,
+                        NoteTxt = $"Activity status changed from { currentActivity.ActivityInstanceStatusTypeCode } to { instance.ActivityInstanceStatusTypeCode }",
```

Comment:
@Maheshbabu28 this should be ActivityInstanceStatusTypeCodeNavigation.Description, so that we don't return an internal code representation when the user is expecting a human-readable representation.

Reply:
@Maheshbabu28 this is the only change that is missing before this can be merged.

Reply:
@devinleighsmith  Done